hw/ip/otp_ctrl/data/BUILD

```diff
@@ -94,6 +94,9 @@ otp_json(
                 # `kSigverifySpxDisabledOtp` and `kSigverifySpxEnabledOtp` in
                 # sw/device/silicon_creator/lib/sigverify/spx_verify.h.
                 "CREATOR_SW_CFG_SIGVERIFY_SPX_EN": otp_hex(0x8d6c8c17),
+                # Use pre-hashed SPX+ (if SPX+ enabled). See the definition of
+                # `hardened_bool_t` in sw/device/lib/base/hardened.h.
+                "CREATOR_SW_CFG_SIGVERIFY_SPX_PREHASH": otp_hex(CONST.HARDENED_FALSE),
```

Comment:
I think we should consult with @moidx on this:  should the SPX+ variant be determined by OTP, or should we put something in the manifest to identify the signature variant?

Reply:
The specification recommends not using the same key in both pre-hashed and pure forms, so using this new OTP setting is ok. 

I prefer to use the existing `ROT_CREATOR_AUTH_CODESIGN_SPX_KEY_CONFIG0` so that we can make this decision at key provisioning time. 

https://github.com/lowRISC/opentitan/blob/6e698b4dfec2f9e61e091cbb9a866197681745f9/hw/ip/otp_ctrl/data/otp_ctrl_mmap.hjson#L430-L433

The key types can be augmented here: 

https://github.com/lowRISC/opentitan/blob/6e698b4dfec2f9e61e091cbb9a866197681745f9/sw/device/silicon_creator/lib/sigverify/spx_key.h#L56-L76

We can consider removing the config fields if we are not planning to use them, but take a look first. It is possible that the config field is part of the key struct used in spx sigverify.

Reply:
As I understand it, the guidance is that one should not use the same key for creating pure vs. pre-hashed signatures.  If we follow that guidance, then like the algorithm variant, then pure vs. pre-hashed tells you how you can use the key.

With that in mind, recall the OTP structure that encodes the key:
```
SPX_KEY_TYPEn: a word associating a key with the TEST, DEV or PROD lifecycle states.
SPX_KEYn: 32 bytes of key material
SPX_KEY_CONFIGn: a word identifying the key algorithm (SHAKE-128s, SHA2-128s, etc).
```

I think we can dual-purpose the `CONFIGn` word to encode both the algorithm and key-usage constraint.  That is to say:
```
enum key_config_t {
  kShake_128s_pure,
  kShake_128s_prehashed,
  kSha2_128s_pure,
  kSha2_128s_prehashed,
  ...
}
```

Then plumb the [`config_id` field](https://github.com/lowRISC/opentitan/blob/6e698b4dfec2f9e61e091cbb9a866197681745f9/sw/device/silicon_creator/rom/sigverify_key_types.h#L223-L236) through and use it to call the verify function appropriately.

Reply:
It is almost there... 

https://github.com/lowRISC/opentitan/blob/6e698b4dfec2f9e61e091cbb9a866197681745f9/sw/device/silicon_creator/rom/sigverify_key_types.h#L223-L236